xds/internal/resolver/xds_resolver_test.go

```diff
@@ -1169,257 +655,128 @@ func (s) TestResolverRemovedResource(t *testing.T) {
-	defer func(oldNewWRR func() wrr.WRR) { newWRR = oldNewWRR }(newWRR)
-	newWRR = testutils.NewTestWRR
-
-	mgmtServer, err := e2e.StartManagementServer(e2e.ManagementServerOptions{})
-	if err != nil {
-		t.Fatal(err)
-	}
-	defer mgmtServer.Stop()
-
-	// Create a bootstrap configuration specifying the above management server.
-	nodeID := uuid.New().String()
-	cleanup, err := xdsbootstrap.CreateFile(xdsbootstrap.Options{
-		NodeID:    nodeID,
-		ServerURI: mgmtServer.Address,
-	})
-	if err != nil {
-		t.Fatal(err)
-	}
-	defer cleanup()
-
-	const serviceName = "my-service-client-side-xds"
-	tcc, rClose := buildResolverForTarget(t, resolver.Target{URL: *testutils.MustParseURL("xds:///" + serviceName)})
-	defer rClose()
-
-	ldsName := serviceName
-	rdsName := "route-" + serviceName
-	// Configure the management server with a good listener resource and a
-	// route configuration resource.
-	resources := e2e.UpdateOptions{
-		NodeID:    nodeID,
-		Listeners: []*v3listenerpb.Listener{e2e.DefaultClientListener(ldsName, rdsName)},
-		Routes: []*v3routepb.RouteConfiguration{e2e.RouteConfigResourceWithOptions(e2e.RouteConfigOptions{
-			RouteConfigName:      rdsName,
-			ListenerName:         ldsName,
-			ClusterSpecifierType: e2e.RouteConfigClusterSpecifierTypeWeightedCluster,
-			WeightedClusters:     map[string]int{"A": 75, "B": 25},
-		})},
-		SkipValidation: true,
-	}
-	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
-	defer cancel()
-	if err := mgmtServer.Update(ctx, resources); err != nil {
-		t.Fatal(err)
-	}
-
-	// Read the update pushed by the resolver to the ClientConn.
-	gotState, err := tcc.stateCh.Receive(ctx)
-	if err != nil {
-		t.Fatalf("Timeout waiting for an update from the resolver: %v", err)
-	}
-	rState := gotState.(resolver.State)
-	if err := rState.ServiceConfig.Err; err != nil {
-		t.Fatalf("Received error in service config: %v", rState.ServiceConfig.Err)
-	}
-	cs := iresolver.GetConfigSelector(rState)
-	if cs == nil {
-		t.Fatal("Received nil config selector in update from resolver")
-	}
-
-	// Make RPCs are verify WRR behavior in the cluster specifier.
-	picks := map[string]int{}
-	for i := 0; i < 100; i++ {
-		res, err := cs.SelectConfig(iresolver.RPCInfo{Context: ctx, Method: "/service/method"})
-		if err != nil {
-			t.Fatalf("cs.SelectConfig(): %v", err)
+	var state resolver.State
+	select {
+	case <-ctx.Done():
+		t.Fatalf("Timeout waiting for an update from the resolver: %v", ctx.Err())
+	case state = <-stateCh:
+		if err := state.ServiceConfig.Err; err != nil {
+			t.Fatalf("Received error in service config: %v", state.ServiceConfig.Err)
+		}
+		wantSCParsed := internal.ParseServiceConfig.(func(string) *serviceconfig.ParseResult)("{}")
+		if !internal.EqualServiceConfigForTesting(state.ServiceConfig.Config, wantSCParsed.Config) {
+			t.Fatalf("Got service config:\n%s \nWant service config:\n%s", cmp.Diff(nil, state.ServiceConfig.Config), cmp.Diff(nil, wantSCParsed.Config))
```

Comment:
The existing helper `verifyUpdateFromResolver` does this and a little more, i.e retrieves the config selector. The config selector in this case though is `nil` and the helper doesn't like that because the config selector being `nil` is not the common case. I initially had two helpers for this:
- verifyUpdateFromResolver
- verifyUpdateFromResolverAndReturnConfigSelector

But since the latter is the more common case, I decided to get rid of one of the helpers. This is not repeated too many times in the code (2 to be exact). So, I would like to leave it as such, until we have more repeated code and we can pull it into a separate helper. 

Thanks.

Reply:
Alright sounds good.